src/go/configgenerator/filtergen/util.go

```diff
@@ -48,8 +48,10 @@ func MethodToSelector(api *apipb.Api, method *apipb.Method) string {
 
 // MethodToCORSSelector gets the corresponding autogenerated CORS selector for
 // an OP API and Method.
-func MethodToCORSSelector(api *apipb.Api, method *apipb.Method) string {
-	return api.GetName() + corsOperationDelimiter + method.GetName()
+func MethodToCORSSelector(api *apipb.Api, method *apipb.Method, opts options.ConfigGeneratorOptions) string {
+	// Override existing corsOperationDelimiter
+	corsOperationDelimiter = opts.CorsOperationDelimiter
```

Comment:
I like moving `CorsOperationDelimiter` to the options struct. But I don't like continuing to use global variable `corsOperationDelimiter` and overriding it.

It's causing hidden dependencies between functions via global variables, as `MethodToCORSSelector` has side effects. A developer needs to remember to call `CORSSelectorToSelector` due to the global variable change.

Can you find a way to propagate CorsOperationDelimiter into `CORSSelectorToSelector`? You can store as struct variable in filter generators and then pass it down. It makes deps clearer

Reply:
Passing down a structure seems a really good way to avoid dependency between two functions. Thanks for the direction! :)

Seems like we also don't require `CorsOperationDelimiter` as part of options struct anymore. But I am keeping it same as it can help developers override it independent of `HealthCheckAutogeneratedOperationPrefix`.